starlette/routing.py

```diff
@@ -557,17 +560,20 @@ async def __call__(self, scope: Scope, receive: Receive, send: Send) -> None:
 
         partial = None
 
+        scope["routes"] = []
```

Comment:
Hmm, perhaps I'm misunderstanding the aim of this — assuming we want `routes` to contain all `BaseRoute` items (mounts, routes) that have matched while going through the routing system, it seems that currently it will only ever contain a single item, i.e. the first one that matched? In case of a route under a `Mount`-ed app, it will be the `Mount` (even if the mounted app is actually a Starlette app perhaps with its own router), and in case of a regular route then it will be that `Route`. Correct?

Reply:
Basing my judgement off https://github.com/encode/starlette/pull/804#issuecomment-590261001 here - Not entirely sure that's something that's actually possible at all, given the ASGI-interoperable design aspects of Starlette.

For example, if you have…

```python
async def other_non_starlette_app(scope, receive, send):
    ...

app = Starlette(routes=[Mount("/subapp", other_non_starlette_app)])
```

Then in this example context using `scope["routes"][-1]` as a way to get the "route that actually responds to the request" might be a bit confusing — it's only guaranteed to be the _last route within this app_ that handles the request _at least partly_.

Don't know if we want to play on words here, or if for the use case you're describing (manipulating route information for telemetry purposes) having partial route info in some cases and marking it as a limitation would be enough?

Reply:
> it seems that currently it will only ever contain a single item, i.e. the first one that matched?

Yes, you are right! I didn't proceed with the PR because wanted to discuss the best strategy with the maintainers before and wasn't sure this was it.

Reply:
> Not entirely sure that's something that's actually possible at all, given the ASGI-interoperable design aspects of Starlette.

An example would be subapps on a FastAPI app? If so, I posted some code on a comment here that works by iterating the routes and searching for a match, but I don't think the problem you described would happen with FastAPI :/